src/components/ConfigEditor/AzureCredentialsForm.tsx

```diff
@@ -0,0 +1,294 @@
+        }
+      }
+    }
+  };
+
+  const onAuthTypeChange = (selected: SelectableValue<AzureAuthType>) => {
+    if (onCredentialsChange) {
+      setSubscriptions([]);
+      const updated: AzureCredentials = {
+        ...credentials,
+        authType: selected.value || 'msi',
+        defaultSubscriptionId: undefined,
+      };
+      onCredentialsChange(updated);
+    }
+  };
+
+  const onAzureCloudChange = (selected: SelectableValue<string>) => {
+    if (onCredentialsChange && (credentials.authType === 'clientsecret' || credentials.authType === 'clientsecret-obo')) {
+      setSubscriptions([]);
+      const updated: AzureCredentials = {
+        ...credentials,
+        azureCloud: selected.value,
+        defaultSubscriptionId: undefined,
+      };
+      onCredentialsChange(updated);
+    }
+  };
+
+  const onTenantIdChange = (event: ChangeEvent<HTMLInputElement>) => {
+    if (onCredentialsChange && (credentials.authType === 'clientsecret' || credentials.authType === 'clientsecret-obo')) {
+      setSubscriptions([]);
+      const updated: AzureCredentials = {
+        ...credentials,
+        tenantId: event.target.value,
+        defaultSubscriptionId: undefined,
+      };
+      onCredentialsChange(updated);
+    }
+  };
+
+  const onClientIdChange = (event: ChangeEvent<HTMLInputElement>) => {
+    if (onCredentialsChange && (credentials.authType === 'clientsecret' || credentials.authType === 'clientsecret-obo')) {
+      setSubscriptions([]);
+      const updated: AzureCredentials = {
+        ...credentials,
+        clientId: event.target.value,
+        defaultSubscriptionId: undefined,
+      };
+      onCredentialsChange(updated);
+    }
+  };
+
+  const onClientSecretChange = (event: ChangeEvent<HTMLInputElement>) => {
+    if (onCredentialsChange && (credentials.authType === 'clientsecret' || credentials.authType === 'clientsecret-obo')) {
+      setSubscriptions([]);
+      const updated: AzureCredentials = {
+        ...credentials,
+        clientSecret: event.target.value,
+        defaultSubscriptionId: undefined,
+      };
+      onCredentialsChange(updated);
+    }
+  };
+
+  const onClientSecretReset = () => {
+    if (onCredentialsChange && (credentials.authType === 'clientsecret' || credentials.authType === 'clientsecret-obo')) {
+      setSubscriptions([]);
+      const updated: AzureCredentials = {
+        ...credentials,
+        clientSecret: '',
+        defaultSubscriptionId: undefined,
+      };
+      onCredentialsChange(updated);
+    }
+  };
+
+  const onSubscriptionChange = (selected: SelectableValue<string> | undefined) => {
+    if (onCredentialsChange) {
+      const updated: AzureCredentials = {
+        ...credentials,
+        defaultSubscriptionId: selected?.value,
+      };
+      onCredentialsChange(updated);
+    }
+  };
+
+  return (
+    <div className="gf-form-group">
+      {(authTypeOptions.length > 1) && (
+        <div className="gf-form-inline">
+          <div className="gf-form">
```

Comment:
if we are going to rewrite this, please use the new componetns from `@grafana/ui`. Check the Azure Monitor config editor to see how it's done there (with `InlineField` components).

Reply:
Done.